File: src/main/java/org/arbor/gtnn/GTNN.java
package org.arbor.gtnn;

import com.gregtechceu.gtceu.api.machine.MachineDefinition;
import com.gregtechceu.gtceu.utils.FormattingUtil;
import com.mojang.logging.LogUtils;
import net.minecraft.resources.ResourceLocation;
import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.eventbus.api.IEventBus;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
import org.arbor.gtnn.config.ConfigHandler;
import org.arbor.gtnn.init.CommonProxy;
import org.slf4j.Logger;

@Mod(GTNN.MODID)
public class GTNN {

    public static final String MODID = "gtnn";
    public static final Logger LOGGER = LogUtils.getLogger();

    public GTNN() {
        MinecraftForge.EVENT_BUS.register(this);
        IEventBus eventBus = FMLJavaModLoadingContext.get().getModEventBus();
        eventBus.addGenericListener(MachineDefinition.class, GTNNRegistries::registerMachine);
        CommonProxy.init();
    }

    public static ConfigHandler.ClientConfigs getClientConfig() {
        return ConfigHandler.INSTANCE.Client;
    }

    public static ConfigHandler.ServerConfigs getServerConfig() {
        return ConfigHandler.INSTANCE.Server;
    }

    public static ResourceLocation id(String path) {
        return new ResourceLocation(MODID, FormattingUtil.toLowerCaseUnder(path));
    }

}

File: src/main/kotlin/dev/arbor/gtnn/GTNN.kt
package dev.arbor.gtnn

import com.gregtechceu.gtceu.utils.FormattingUtil
import dev.arbor.gtnn.config.GTNNConfigHandler
import dev.arbor.gtnn.init.ClientProxy
import dev.arbor.gtnn.init.CommonProxy
import net.minecraft.resources.ResourceLocation
import net.minecraftforge.fml.DistExecutor
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger
import java.util.function.Supplier


object GTNN {
    const val MOD_ID = "gtnn"

    val LOGGER: Logger by lazy { LogManager.getLogger(MOD_ID) }

    @JvmStatic
    fun init() {
        DistExecutor.unsafeRunForDist({
            Supplier {
                ClientProxy()
            }
        }, {
            Supplier {
                CommonProxy()
            }
        }).init()
    }

    fun getClientConfig(): GTNNConfigHandler.ClientConfigs {
        return GTNNConfigHandler.INSTANCE.Client
    }

    fun getServerConfig(): GTNNConfigHandler.ServerConfigs {
        return GTNNConfigHandler.INSTANCE.Server
    }

    fun id(path: String): ResourceLocation {
        return ResourceLocation(MOD_ID, FormattingUtil.toLowerCaseUnder(path))
    }
}

